### DISCUSSION TITLE
[Discussion] New possible card ideas

### DISCUSSION THREAD

**Discussion by HPaulson** [anuraghazra/github-readme-stats#945] [Ideas]
Discussion of new possible card ideas & options for existing cards!
--
### Possible Card Ideas:

**Compact Stats Card**
- Having a compact stats card, so it can fit side-by-side with the compact language card, and decrease unnecessary whitespace and wording in the card
- [Concept 1](https://cdn.smc.wtf/NrQ.png), [Concept 2](https://user-images.githubusercontent.com/20476002/88905874-15537480-d257-11ea-91ef-e0b6d319f5aa.png)

**Pie chart language card**
- Having an additional option for the language card, which would format the languages into a pie chart shape rather than the traditional github style, or the default language card style
- [Concept](https://user-images.githubusercontent.com/20476002/88953719-a0edf500-d299-11ea-983d-747d078deb12.png)

**Comment by Ryyn-Sd** [comment]
I'll work on it!

**Comment by Angelk90** [comment]
@HPaulson , @Ryyn-Sd : 
An even more compact shape.
![68747470733a2f2f63646e2e736d632e7774662f4e72512e706e67](https://user-images.githubusercontent.com/20476002/88905874-15537480-d257-11ea-91ef-e0b6d319f5aa.png)

**Comment by nathanchu** [comment]
Or the rank circle could be hidden to make room for the stats. I do like your idea though @Angelk90.

**Comment by HPaulson** [comment]
@Angelk90 Yup! Even better!👍

@nthnchu Keep in mind, the idea is to have both cards be able to go side by side, not for it to be as small as possible. Ideally we'd be able to have the same information displayed, but without unless white space to compact the image.

**Comment by Angelk90** [comment]
@HPaulson : 

![Schermata 2020-07-30 alle 19 08 16 (1)](https://user-images.githubusercontent.com/20476002/88953719-a0edf500-d299-11ea-983d-747d078deb12.png)

What do you think of such a graph instead of what you see below.

![68747470733a2f2f63646e2e736d632e7774662f4e72512e706e67](https://user-images.githubusercontent.com/20476002/88905874-15537480-d257-11ea-91ef-e0b6d319f5aa.png)

**Comment by HPaulson** [comment]
@Angelk90 Yeah that looks kinda bad in my opinion... The current compact lang graph that looks just like github's is very very nice, and this pie chart kinda loses the pleasing ascetic of the current compact one. Additionally, this concept pie chart you have only displays the rank, which as I see it, is the most useless information on the card compared to the actual stats... Personally I really like your original  [concept](https://user-images.githubusercontent.com/47159695/88954307-2fe21980-d268-11ea-95c5-161b10ae8626.png) the best. Overall I think mixing the two graphs isn't a great idea either, since they both display different information that's good in it's own way. Especially since some people may just want a compact stats card, but not want it mixed with languages.

I'm curious what other people's thoughts on this pie chart are, though!

**Comment by anuraghazra** [comment]
> @HPaulson :
> 
> ![Schermata 2020-07-30 alle 19 08 16 (1)](https://user-images.githubusercontent.com/20476002/88953719-a0edf500-d299-11ea-983d-747d078deb12.png)
> 

I like this Pie chart, but i don't think it would look good if we mix it up with stats card, maybe we can make it a new layout option in the Languages card `&layout=pie-chart`

**Comment by anuraghazra** [comment]
> @HPaulson , @Ryyn-Sd :
> An even more compact shape.
> ![68747470733a2f2f63646e2e736d632e7774662f4e72512e706e67](https://user-images.githubusercontent.com/20476002/88905874-15537480-d257-11ea-91ef-e0b6d319f5aa.png)

Issue with this compact card is people would get really confused with just the numbers, and it also looks too much cramped IMO. 

@HPaulson's original card is better because it shows some context of what those numbers means.

**Comment by HPaulson** [comment]
@anuraghazra 
Regarding [This Comment](https://github.com/anuraghazra/github-readme-stats/issues/244#issuecomment-666547642) - I'm pretty sure the idea behind the pie chart was to mix the stats & lang cards... I think this since the pie chart itself is langs, and the center uses the rank as the stats card does. If this was the intention, I think it's kinda poor. However just having a pie chart for langs isn't a bad idea at all, I'd also like to point out that the current compact lang card is a pie chart, just in a straight line instead of the traditional pie shape - But shows the same information in the exact same way, just a different shape.
![Img](https://cdn.smc.wtf/9Pp.png)
#
Also for [This Comment](https://github.com/anuraghazra/github-readme-stats/issues/244#issuecomment-666549087), I think that the numbers appear more clean actually... Since most github users are going to know the icons meaning, since they are used around the platform and are quite universal here. Only one that could get confusing possibly is the commit icon in my opinion. I think it's a bit cleaner without the words however since they are all aligned where the words have different lengths and looks kinda strange when compacted I think. Both work just fine, though. Finally, I do agree that It looks a bit crammed, not sure how we could fix this without adding unnecessary white space and taking away it's compact-ability.

**Comment by HPaulson** [comment]
Also, this may require another issue - Since it's a bit off-topic, but since it's along the lines of reducing space in the stats card, what are everyone's thoughts on adding an additional option to the stats card to hide the rank? Currently you're able to hide any of the specific stats on the left, but you aren't able to hide the rank circle. I feel like you should be able to, since, it just makes sense if you can hide the other info.

**Comment by anuraghazra** [comment]
> Currently you're able to hide any of the specific stats on the left, but you aren't able to hide the rank circle.

You can hide the rank circle `&hide_rank=true`

**Comment by anuraghazra** [comment]
> I think this since the pie chart itself is langs, and the center uses the rank as the stats card does. If this was the intention, I think it's kinda poor.

No, I'm not talking about the centered rank on the pie chart. Just the pie chart in the langs card.

**Comment by HPaulson** [comment]
> You can hide the rank circle `&hide_rank=true`

Ah okay, I'll mark my comment as off-topic then haha. Didn't see that option in docs, sorry about that!

**Comment by HPaulson** [comment]
> No, I'm not talking about the centered rank on the pie chart. Just the pie chart in the langs card.

Yeah as I mentioned later in that comment-
> However just having a pie chart for langs isn't a bad idea at all

What I meant there was just that it seemed the intention of the concept was to merge the rank from stats & lang cards, which I thought was a bit poor. Having just a pie chart for langs is a great idea like I mentioned, however I'll mention again that the current compact one is a pie chart just in a line and not a circle, so it may be a bit repetitive. However if people prefer the circle shape, then it's a great idea. Having more options isn't a bad thing! All I was saying is that mixing the two I think is poor, but if that isn't the intention, than I think that could be a great idea too!

**Comment by HPaulson** [comment]
Also, I have some other possible card ideas that I think could possibly excel this project... Should I make a new issue for them or just change the title here to "Possible card ideas" and leave it as more of an open discussion for more possible ideas, rather than just the compact stats card, since we're already talking about other card ideas such as the pie chart?

**Comment by anuraghazra** [comment]
> Also, I have some other possible card ideas that I think could possibly excel this project... Should I make a new issue for them or just change the title here to "Possible card ideas" and leave it as more of an open discussion for more possible ideas, rather than just the compact stats card, since we're already talking about other card ideas such as the pie chart?


Of course do it!

**Comment by anuraghazra** [comment]
> so it may be a bit repetitive

No it won't be repetitive, it would be just a stylistic choice of which type of design users would like to use.

**Comment by HPaulson** [comment]
Additionally, I had an idea about having a card to show recent activity and contributions - I used to use [this repo](https://github.com/JasonEtco/activity-box) to update a pinned gist to do so, however I think that with the addition of readme it could be much much more improved by being an image, rather than being text in a gist; The design could be much more improved, and it could be a cool card to add! What are everyone's thoughts on something like this? 

Also, I'm going to edit the initial comment on this thread to include the ideas we've shared so far.

**Comment by HPaulson** [comment]
Just a quick update on this, I did actually make some changes to the stats card to see how this would come out, and I think it looks great!

@anuraghazra What are your thoughts?

<details>
  <summary><b>Changes</b></summary>
  
  ```diff
- 34      <text class="stat bold" ${labelOffset} y="12.5">${label}:</text>

+ 35      ${labelOffset}

- 37      x="${shiftValuePos ? (showIcons ? 200 : 170) : 150}" 

+ 40      <text class="stat bold" x="60" y="12.5">${label}</text>

- 85      label: "Total Stars",
+ 85      label: "Stars Received",

- 91      label: `Total Commits${
+ 91      label: `Commits${

- 99      label: "Total PRs",
+ 99      label: "PRs",

- 105     label: "Total Issues",
+ 105     label: "Issues",

- 111     label: "Contributed to",

- 134     hide_rank ? 0 : 150
+ 134     hide_rank ? 0 : 75

- 141     hide_rank ? 0 : 75
+ 141     transform="translate(300, ${height / 2 - 50})">

- 171     width: 495,
+ 171     width: 350,
  ```
</details>

![](https://cdn.smc.wtf/hJA.png)

**Comment by Angelk90** [comment]
@anuraghazra : 
1) Is there any news regarding the pie chart?
2) Is there any news regarding the possibility of having only the icons in the github stats card without having the labels?

**Comment by HPaulson** [comment]
@Angelk90 This thread is just a basic discussion of different possible card ideas- if you want one added, I'd suggest opening a PR & creating it!

**Comment by cmcodes1** [comment]
Height of Top Languages Card should be equal to the height of GitHub Stats Card, so that it looks symmetrical when aligned side-by-side.

**Comment by Ryyn-Sd** [comment]
I decided to not do it

**Comment by rickstaa** [comment]
I will put this here for future reference. This discussion is meant to be a place where the community can ask for feedback about a possible card idea from the @github-readme-stats core team and the community before creating [a feature request](https://github.com/anuraghazra/github-readme-stats/issues) or [PR](https://github.com/anuraghazra/github-readme-stats/pulls).

We have not yet implemented any of the features requested by @Angelk90 in https://github.com/anuraghazra/github-readme-stats/discussions/945#discussioncomment-489459. 

- For the compact stats card, without any text, I think only the symbols is too unclear. This feature request is related to #1170. We will merge #1334 in the future to provide this option.

- I do like the pie chart version of the language card. I just opened a feature request for this #1650. Feel free to create a PR for it.
